ducktape/tests/test.py

```diff
@@ -143,14 +160,18 @@ def results_dir(self):
             d = os.path.join(d, self.cls.__name__)
         if self.function is not None:
             d = os.path.join(d, self.function.__name__)
+
+        if self.injected_args is not None:
+            # The function has arguments injected; append these to the id
+            parameters = ".".join(["%s=%s" % (k, self.injected_args[k]) for k in self.injected_args])
```

Comment:
Not sure there's a good solution for this, but this may need some filtering if it's going to be used as part of a path. At a bare minimum, `os.sep` needs filtering/escaping or it'll create additional subdirectories. There are probably other characters that'll be odd to include.

Not sure if this is a realistic problem thought -- in practice I assume most will be simple number or string settings. On the other hand, settings like zkconnect with a chroot (which you might use for testing use of chroot/multitenancy) would include path separators...